src/sqlfluff/core/linter/linted_file.py

```diff
@@ -330,10 +330,11 @@ def fix_string(self) -> Tuple[Any, bool]:
                 source_str=self.templated_file.source_str[source_slice],
             )
 
-            # Deal with the easy case of only literals
-            if set(local_type_list) == {"literal"}:
+            # Deal with the easy cases of 1) New code at end 2) only literals
+            if not local_type_list or set(local_type_list) == {"literal"}:
```

Comment:
Prior to this fix, the linter was discarding the fix if we added a newline at the end of the file, immediately following a templated region.